### ISSUE TITLE
Emacs error loading: wrong number of arguments

### ISSUE THREAD

**Issue by MisterDA** [ocaml/merlin#1234]
```
$ emacs --version 
GNU Emacs 28.0.50
$ ocamlmerlin -version
The Merlin toolkit version v3.4.2-36-gfc2f6e57, for Ocaml 4.11.1
```

I use latest development branch of Emacs patched for pure GTK3 (emacs-pgtk-native-comp-git-28.0.50.146292-1 in Arch Linux). It seems a function has changed the dev branch of Emacs, because I get this crash since a few days. I cannot bisect upstream, the build time is too long, and I lack the Emacs-fu to debug this myself. An idea?
I've installed latest merlin through opam and user-setup.

```
Debugger entered--Lisp error: (wrong-number-of-arguments (3 . 4) 2)
  #f(compiled-function (obsolete-name current-name when &optional docstring) "Set OBSOLETE-NAME's function definition to CURRENT-NAME and mark it obsolete.\n\n(define-obsolete-function-alias \\='old-fun \\='new-fun \"22.1\" \"old-fun's doc.\")\n\nis equivalent to the following two lines of code:\n\n(defalias \\='old-fun \\='new-fun \"old-fun's doc.\")\n(make-obsolete \\='old-fun \\='new-fun \"22.1\")\n\nWHEN should be a string indicating when the function was first\nmade obsolete, for example a date or a release number.\n\nSee the docstrings of `defalias' and `make-obsolete' for more details." #<bytecode -0xd058badb3ebf7bb>)('merlin-project-check 'merlin-configuration-check)
  (define-obsolete-function-alias 'merlin-project-check 'merlin-configuration-check)
  load-with-code-conversion("/home/antonin/.opam/ocaml-base-compiler.4.11.1/sha..." "/home/antonin/.opam/ocaml-base-compiler.4.11.1/sha..." nil t)
  require(merlin)
  (lambda nil (opam-setup-complete) (require 'merlin) (opam-setup-add-ocaml-hook 'merlin-mode) (custom-declare-variable 'ocp-index-use-auto-complete 'nil "Use auto-complete with ocp-index (disabled by defa..." :group 'ocp_index) (custom-declare-variable 'merlin-ac-setup ''easy "Use auto-complete with merlin (enabled by default ..." :group 'merlin-ac) (define-key merlin-mode-map (kbd "C-c <up>") 'merlin-type-enclosing-go-up) (define-key merlin-mode-map (kbd "C-c <down>") 'merlin-type-enclosing-go-down) (set-face-background 'merlin-type-face "skyblue"))()
  funcall((lambda nil (opam-setup-complete) (require 'merlin) (opam-setup-add-ocaml-hook 'merlin-mode) (custom-declare-variable 'ocp-index-use-auto-complete 'nil "Use auto-complete with ocp-index (disabled by defa..." :group 'ocp_index) (custom-declare-variable 'merlin-ac-setup ''easy "Use auto-complete with merlin (enabled by default ..." :group 'merlin-ac) (define-key merlin-mode-map (kbd "C-c <up>") 'merlin-type-enclosing-go-up) (define-key merlin-mode-map (kbd "C-c <down>") 'merlin-type-enclosing-go-down) (set-face-background 'merlin-type-face "skyblue")))
  (progn (funcall (symbol-function (cdr tool))))
  (if (member (car tool) opam-tools-installed) (progn (funcall (symbol-function (cdr tool)))))
  (while --dolist-tail-- (setq tool (car --dolist-tail--)) (if (member (car tool) opam-tools-installed) (progn (funcall (symbol-function (cdr tool))))) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (let ((--dolist-tail-- opam-tools) tool) (while --dolist-tail-- (setq tool (car --dolist-tail--)) (if (member (car tool) opam-tools-installed) (progn (funcall (symbol-function (cdr tool))))) (setq --dolist-tail-- (cdr --dolist-tail--))))
  opam-auto-tools-setup()
  load-with-code-conversion("/home/antonin/.emacs.d/opam-user-setup.el" "/home/antonin/.emacs.d/opam-user-setup.el" nil t)
  require(opam-user-setup "~/.emacs.d/opam-user-setup.el")
  load-with-code-conversion("/home/antonin/.emacs.d/init.el" "/home/antonin/.emacs.d/init.el" t t)
  load("/home/antonin/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode -0xb99a4c95394408a>) #f(compiled-function () #<bytecode 0xc0853d4c320650c>) t)
  command-line()
  normal-top-level()
```

**Comment by Lelio-Brun** [comment]
I think I found the culprit : 
http://git.savannah.gnu.org/cgit/emacs.git/commit/lisp/emacs-lisp/byte-run.el?id=32c6732d16385f242b1109517f25e9aefd6caa5c
The `when` argument of the function `define-obsolete-function-alias` has recently been made mandatory.

The error comes from eg. here, where obviously the `when` string is missing : https://github.com/ocaml/merlin/blob/722bcc6bce06b5b10d8f14f9b7feac76ee0ef95f/emacs/merlin.el#L1821

**Comment by MisterDA** [comment]
Cool! Thanks for the diagnostic and the fix!

**Comment by basus** [comment]
Any idea when a version with this fix will be available on OPAM? I'm assuming it's part of 4.0, but OPAM seems to still be at 3.4.2.

**Comment by trefis** [comment]
It should reach opam in the next couple of days.
